sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TableLookupCacheSuite.scala

```diff
@@ -29,7 +29,7 @@ import org.scalatest.matchers.must.Matchers
 import org.apache.spark.sql.catalyst.TableIdentifier
 import org.apache.spark.sql.catalyst.catalog.{CatalogDatabase, CatalogStorageFormat, CatalogTable, CatalogTableType, ExternalCatalog, InMemoryCatalog, SessionCatalog}
 import org.apache.spark.sql.catalyst.dsl.plans._
-import org.apache.spark.sql.connector.{InMemoryTable, InMemoryTableCatalog}
+import org.apache.spark.sql.connector.{InMemoryTable, V2InMemoryCatalog}
```

Comment:
Can we avoid changing the `InMemoryTableCatalog` to keep changes smaller? It seems like we could create a separate `InMemoryFunctionCatalog` instead.

Reply:
Good point. I removed most of the changes, but still kept the `InMemoryCatalog` though, thinking that it would be useful to have a test catalog with both tables and functions (similar to the V1 `InMemoryCatalog`). It is currently only used in `DataSourceV2FunctionSuite` and `CatalogSuite`.